processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -129,7 +128,7 @@ private Builder setupMethodWithMapping(Method sourceMethod) {
             for ( Parameter sourceParameter : method.getSourceParameters() ) {
                 unprocessedSourceParameters.add( sourceParameter );
 
-                if ( sourceParameter.getType().isPrimitive() ) {
+                if ( sourceParameter.getType().isPrimitive() || sourceParameter.getType().isArrayType() ) {
```

Comment:
This is and line 573 are the actual fixes for the NPE right?

Reply:
Exactly! Looks for me like it was tried to find getter/setters even if it is an array. This will be skipped there for primitives and thus I also added the skip for arrays. I am not so deep in this internal stuff, that's the reason I asked that someone else should check this as well. At least I haven't destroyed a different testcase. So hopefully I analyzed and fixed this correctly.

Reply:
The analysis was great. This is exact place where the fix should come. I was just curious whether there was something additional, but it seems not